src/wallets/TransactionRow.tsx

```diff
@@ -157,14 +157,16 @@ const Confirmations = ({transaction: {blockNumber}}: TransactionCellProps): JSX.
   if (
     blockNumber === null ||
     walletState.walletStatus !== 'LOADED' ||
-    walletState.syncStatus.mode !== 'online'
+    walletState.syncStatus.mode === 'offline'
   )
     return <></>
 
-  const confirmations = walletState.syncStatus.highestKnownBlock - blockNumber
+  const highestBlock =
+    walletState.syncStatus.mode === 'synced'
+      ? walletState.syncStatus.currentBlock
+      : walletState.syncStatus.highestKnownBlock
+  const confirmations = highestBlock - blockNumber
 
-  // Didn't find exact conditions, but sometimes highest known block is not there?
```

Comment:
Not relevant anymore? AFAIK there was a bug which caused `highestKnownBlock` to be 0, no matter the value of `currentBlock`.

Reply:
Hm...I thought the comment is referring to the code below...explaining why we need to check whether `confirmations` is not `NaN`. But the case where it's `undefined` (not 0) is exactly handled in this PR. And the case where it could be 0 is not really handled anywhere (and I didn't really encounter it).